docs/guides/run-elasticsearch-local.rst

```diff
@@ -0,0 +1,99 @@
+==========================================
+Enabling Elasticsearch on the local server
+==========================================
+
+Read the Docs has been using Elasticsearch which is a platform for distributed search and analysis of data in real time. To enable the search feature on your local installation, you need to install the elasticsearch locally and run the Elastic server. 
+
+Installation has been mainly divided into following steps.
+
+1. Installing Java
+------------------
+
+Elasticsearch requires JAVA 8 or later. Use `Oracle official documentation <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`_. 
+or opensource distribution like `OpenJDK <http://openjdk.java.net/install/>`_.
+
+After installing java, verify the installation by,::
+
+	$ java -version
+
+The result should be something like this::
+
+	openjdk version "1.8.0_151"
+	OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12)
+	OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
+
+
+2. Downloading and installing Elasticsearch
+-------------------------------------------
+
+Elasticsearch can be downloaded directly from elastic.co in zip, tar.gz, deb, or rpm packages. For Ubuntu, it's best to use the deb (Debian) package which will install everything you need to run Elasticsearch.
+
+RTD currently uses elasticsearch 1.x which can be easily downloaded and installed from the `Official Website 
+<https://www.elastic.co/downloads/>`_.
+
+	1. First download the deb package version 1.3.8 from this `Link <https://www.elastic.co/downloads/past-releases/elasticsearch-1-3-8>`_.
+
+	2. Install the downloaded package by following command 
+		``$ sudo apt install .{path-to-downloaded-file}/elasticsearch-1.3.8.deb``
+
+3. Running Elasticsearch from command line
+------------------------------------------
+
+Elasticsearch is not started automatically after installation. How to start and stop Elasticsearch depends on whether your system uses SysV init or systemd (used by newer distributions). You can tell which is being used by running this command::
+
+	$ ps -p 1	
+
+**Running Elasticsearch with SysV init**
```

Comment:
I'd leave how to run Elasticsearch to users for several reasons:

1. There are multiple init systems and it would be nice to not maintain these commands in our documentation.
2. People may want to add their choice of init system to the documentation and that would also increase the maintenance work.
3. They also can use it without using any init systems and package managers:

    ```sh
    $ wget https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch-1.3.8.tar.gz
    $ tar xf elasticsearch-1.3.8.tar.gz
    $ cd elasticsearch-1.3.8/
    $ ./bin/elasticsearch
    ```

   I wonder if we should document this method instead since it makes installing and running Elasticsearch without having to deal with different init systems, package managers, and distros.

Reply:
First I thought of writing exactly the same as mentioned by you (i.e directly downloading the tar file), it was previously covered in #3852 . As the said method has an advantage of not having to maintain the init systems, we have a disadvantage of having to start the elasticsearch server each time we need to run our RTD server locally.